MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.mm

```diff
@@ -812,23 +871,32 @@
 
 	} else if (!forTessellation && stage == kMVKGraphicsStageRasterization) {
         encodeBindings(kMVKShaderStageVertex, "vertex", fullImageViewSwizzle,
-                       [pipeline](MVKCommandEncoder* cmdEncoder, MVKMTLBufferBinding& b)->void {
+					   [pipeline, isDynamicVertexStride](MVKCommandEncoder* cmdEncoder, MVKMTLBufferBinding& b)->void {
                            // The app may have bound more vertex attribute buffers than used by the pipeline.
                            // We must not bind those extra buffers to the shader because they might overwrite
                            // any implicit buffers used by the pipeline.
                            if (pipeline->isValidVertexBufferIndex(kMVKShaderStageVertex, b.index)) {
-                               if (b.isInline) {
-                                   cmdEncoder->setVertexBytes(cmdEncoder->_mtlRenderEncoder,
-                                                              b.mtlBytes,
-                                                              b.size,
-                                                              b.index);
+							   NSUInteger mtlStride = isDynamicVertexStride ? b.stride : MTLAttributeStrideStatic;
+							   if (b.isInline) {
+								   [cmdEncoder->_mtlRenderEncoder setVertexBytes: b.mtlBytes
+																		  length: b.size
+#if MVK_XCODE_15
+																 attributeStride: mtlStride
+#endif
```

Comment:
What happens if MoltenVK is built with Xcode 15, but deployed on a system older than macOS 14/iOS 17?

Reply:
🤦🏻‍♂️ Thanks! Good catch! 

I clearly tried to outsmart myself there, with minimizing the changes.

I've consolidated the changes into `MVKCommandEncoder::encodeVertexAttributeBuffer()`, to avoid rampant repeated spaghetti code.